azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/internal_client.py

```diff
@@ -136,3 +136,22 @@ def from_authentication_provider(cls, authentication_provider, transport_name):
                 "No specific transport can be instantiated based on the choice."
             )
         return cls(authentication_provider, transport)
+
+
+# This is where we add our _async functions to InternalClient.  We do this at runtime
+# by adding them to the InternalClient prototype, using what the python community calls
+# "monkeypatching".  If we're running on python 2.7, the import will throw a SyntaxError
+# and we'll just carry on without the _async functions.
+#
+# Don't worry, adding them to InternalClient at runtime will also add them to DeviceClient
+# and ModuleClient.  Python has a module cache like node, so the first time anyone imports
+# internal_client.py (to use it for a base class, for instance), it will monkeypatch it here
+# and use the patched class definition for anyone else who imports internal_client.py
```

Comment:
As soon as possible please sync up with Oliva on the docs generation mechanism to see how that looks like in the docs